go-controller/pkg/util/pod_annotation.go

```diff
@@ -395,6 +379,20 @@ func PodNadNames(pod *v1.Pod, netinfo NetInfo) ([]string, error) {
 	return nadNames, nil
 }
 
+func GetPrimaryNetworkNADNamesForNamespaceFromNetInfo(namespace string, netinfo NetInfo) ([]string, error) {
+	for _, nadName := range netinfo.GetNADs() {
+		ns, _, err := cache.SplitMetaNamespaceKey(nadName)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing nad name %s from network %s: %v", nadName, netinfo.GetNetworkName(), err)
+		}
+		if ns != namespace {
+			continue
+		}
+		return []string{nadName}, nil
```

Comment:
This is based on two assumptions:

- There is a single NAD per namespace for primary networks. However, there is nothing currently that prevents this from happening.
- We are aware of the aware of the primary network of a pod. However this might be reacey.

The first issue should be handled in some way by the NAD controller. It might do that through `GetActiveNetworkForNamespace`. For the second issue, `GetActiveNetworkForNamespace` offers a bit more of protection since it will check against the UDN CRDs as well.

I don't want to hold this because of that. But I am entertaining the idea that we could just implement `GetActiveNetworkForNamespace` in NetInfo that would just end up calling the nad controller `GetActiveNetworkForNamespace` which is feasible since the nad controller is the one building those NetInfo after all.

Reply:
I like the idea of having `GetActiveNetworkForNamespace` inside netInfo.